app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,46 @@
+module SessionsHelper
+
+  # Logs in the given user.
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  # Remembers a user in a persistent session.
+  def remember(user)
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  # Returns the user corresponding to the remember token cookie.
+  def current_user
```

Comment:
Can you break this up into smaller pieces please? This many conditionals (especially nested) makes me think that there is too much going on in the method.